debug_scripts/estimate_epoch_start_time.py

```diff
@@ -119,6 +119,53 @@ def find_epoch_for_timestamp(future_epochs, voting_timestamp):
     return len(future_epochs)
 
 
+def find_best_voting_hour(voting_date_str, future_epochs, target_timezone):
+    WORKING_HOURS_START = 8  # 8:00 UTC
+    WORKING_HOURS_END = 22  # 22:00 UTC
+
+    valid_hours = []
+
+    for hour in range(24):
+        # Construct datetime for each hour of the voting date
+        voting_datetime = datetime.strptime(
+            f"{voting_date_str} {hour:02d}:00:00", '%Y-%m-%d %H:%M:%S')
+        voting_datetime = target_timezone.localize(voting_datetime)
+        voting_timestamp = voting_datetime.timestamp()
+
+        # Find the epoch T in which the voting date falls
+        epoch_T = find_epoch_for_timestamp(future_epochs, voting_timestamp)
+        if epoch_T <= 0:
+            continue  # Voting date is before the first predicted epoch
+
+        # Calculate when the protocol upgrade will happen (start of epoch T+2)
+        protocol_upgrade_epoch_number = epoch_T + 2
+        if protocol_upgrade_epoch_number > len(future_epochs):
+            print(
+                "Not enough future epochs predicted to determine all the protocol upgrade times."
+            )
+            break
+
+        protocol_upgrade_timestamp = future_epochs[protocol_upgrade_epoch_number
+                                                   - 1]
+        protocol_upgrade_datetime = datetime.fromtimestamp(
+            protocol_upgrade_timestamp)
+        upgrade_hour_utc = protocol_upgrade_datetime.hour
+
+        if WORKING_HOURS_START <= upgrade_hour_utc < WORKING_HOURS_END:
+            valid_hours.append((hour, protocol_upgrade_epoch_number))
+
+    if valid_hours:
+        print(
+            f"\nVoting hours on {voting_date_str} {target_timezone} that result in upgrade during working hours (UTC {WORKING_HOURS_START}:00-{WORKING_HOURS_END}:00):"
```

Comment:
When doing a release, we set the voting date in the `UTC`. It is useful to have this output in the following format: voting date in UTC -> upgrade date in working hours for `target_timezone`.

Reply:
Is it good if I display the upgrade epoch and before that the epochs are listed in the `target_timezone` (as in the example output)?